scipy/optimize/_lap.py

```diff
@@ -0,0 +1,111 @@
+# Wrapper for the Jonker-Volgenant algorithm for solving the rectangular
+# linear sum assignment problem.  The original code was an implementation of
+# the Hungarian algorithm (Kuhn-Munkres) taken from scikit-learn, based on
+# original code by Brian Clapper and adapted to NumPy by Gael Varoquaux.
+# Further improvements by Ben Root, Vlad Niculae, Lars Buitinck,
+# and Peter Larsen.
+#
+# Copyright (c) 2008 Brian M. Clapper <bmc@clapper.org>, Gael Varoquaux
+# Author: Brian M. Clapper, Gael Varoquaux
+# License: 3-clause BSD
+
+import numpy as np
+from . import _jonkervolgenant
+
+
+def linear_sum_assignment(cost_matrix, maximize=False):
+    """Solve the linear sum assignment problem.
+
+    The linear sum assignment problem is also known as minimum weight matching
+    in bipartite graphs. A problem instance is described by a matrix C, where
+    each C[i,j] is the cost of matching vertex i of the first partite set
+    (a "worker") and vertex j of the second set (a "job"). The goal is to find
+    a complete assignment of workers to jobs of minimal cost.
+
+    Formally, let X be a boolean matrix where :math:`X[i,j] = 1` iff row i is
+    assigned to column j. Then the optimal assignment has cost
+
+    .. math::
+        \\min \\sum_i \\sum_j C_{i,j} X_{i,j}
+
+    s.t. each row is assignment to at most one column, and each column to at
```

Comment:
I just added #10344 before seeing this PR, but that becomes relevant here as well: There is some room for improvement in the wording of the old documentation (and thus in this one). In particular, requiring "at most one" assignment means that the optimal solution is the trivial one, which isn't what we want.

Reply:
Thanks @fuglede .  Once your PR is accepted I will merge it here.

Reply:
merged now

Reply:
Thanks @fuglede
@rgommers My git skills are not very sharp.  The changes to `scipy/optimize/_hungarian.py` from #10344 are causing conflicts with the base branch here.  I just want to do
```git -rm scipy/optimize/_hungarian.py```, since I have replaced it with ```scipy/optimize/_lap.py```.  Can I merge `master` into this branch, or does that mess up the list of changes?

Reply:
yes, if you can't easily rebase your branch on top of master, then merging master into this branch is fine